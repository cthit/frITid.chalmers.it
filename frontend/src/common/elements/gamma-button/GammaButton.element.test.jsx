import React from "react";
import { shallow } from "enzyme";
import GammaButton from "./GammaButton.element";

describe("<GammaButton/>", () => {
  test("Shallow render of primary raised <GammaButton/>", () => {
    const wrapper = shallow(
      <GammaButton
        text="This is a button"
        onClick={() => console.log("I have been pressed")}
        primary
        raised
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of secondary disabled <GammaButton/>", () => {
    const wrapper = shallow(
      <GammaButton
        text="This is a button"
        onClick={() => console.log("I have been pressed")}
        secondary
        disabled
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test("Shallow render of raised submit <GammaButton/>", () => {
    const wrapper = shallow(
      <GammaButton text="This is a button" raised submit />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
