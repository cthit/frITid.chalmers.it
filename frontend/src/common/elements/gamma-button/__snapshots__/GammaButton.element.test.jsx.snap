// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GammaButton/> Shallow render of primary raised <GammaButton/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaButton
    onClick={[Function]}
    primary={true}
    raised={true}
    text="This is a button"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "This is a button",
      "color": "primary",
      "disabled": undefined,
      "onClick": [Function],
      "type": "button",
      "variant": "contained",
    },
    "ref": null,
    "rendered": "This is a button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "This is a button",
        "color": "primary",
        "disabled": undefined,
        "onClick": [Function],
        "type": "button",
        "variant": "contained",
      },
      "ref": null,
      "rendered": "This is a button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaButton/> Shallow render of raised submit <GammaButton/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaButton
    raised={true}
    submit={true}
    text="This is a button"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "This is a button",
      "color": "default",
      "disabled": undefined,
      "onClick": undefined,
      "type": "submit",
      "variant": "contained",
    },
    "ref": null,
    "rendered": "This is a button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "This is a button",
        "color": "default",
        "disabled": undefined,
        "onClick": undefined,
        "type": "submit",
        "variant": "contained",
      },
      "ref": null,
      "rendered": "This is a button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<GammaButton/> Shallow render of secondary disabled <GammaButton/> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GammaButton
    disabled={true}
    onClick={[Function]}
    secondary={true}
    text="This is a button"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "This is a button",
      "color": "secondary",
      "disabled": true,
      "onClick": [Function],
      "type": "button",
      "variant": "flat",
    },
    "ref": null,
    "rendered": "This is a button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "This is a button",
        "color": "secondary",
        "disabled": true,
        "onClick": [Function],
        "type": "button",
        "variant": "flat",
      },
      "ref": null,
      "rendered": "This is a button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
